hail/python/hail/methods/relatedness/pc_air.py

```diff
@@ -0,0 +1,190 @@
+        relatives = list(map(lambda index_struct: tuple(index_struct.values()), relatives))
+        relatives = [index_values[0] if len(index_values) == 1 else index_values for index_values in relatives]
+        samples.loc[relatives, 'eta'] -= 1
+
+        # Set eta to 0 for the selected sample
+        samples.at[selected_sample, 'eta'] = 0
+
+
+@typecheck(genotypes=expr_call, relatedness_threshold=expr_numeric, divergence_threshold=expr_numeric)
+def pc_air(
+    genotypes: CallExpression,
+    *,
+    relatedness_threshold: Union[int, float, NumericExpression] = 0.025,
+    divergence_threshold: Union[int, float, NumericExpression] = 0.025,
+):
+    """
+    Perform PC-AiR (principal components analysis in related samples) on the genotypes.
+
+    PC-Air is a robust method for inferring population structure from genome-screen data
+    described `here <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4836868/>`_.
+
+    Notes
+    -----
+    This method uses kinship coefficients to determine whether two samples are related.
+    Two samples are considered related if their kinship coefficient is greater than the relatedness threshold.
+    Otherwise, they are considered unrelated.
+
+    Similarly, this method uses kinship coefficient estimates to determine if two samples are ancestrally divergent.
+    Two samples are considered divergent if their ancestral divergence is less than
+    the negation of the divergence threshold.
+
+    Parameters
+    ----------
+    genotypes : :class:`.CallExpression`
+        A call expression representing the genotype calls.
+    relatedness_threshold : :obj:`int` or :obj:`float` or :class:`.NumericExpression`
+        The relatedness threshold. The default is 0.025.
+    divergence_threshold : :obj:`int` or :obj:`float` or :class:`.NumericExpression`
+        The divergence threshold. The default is 0.025.
+
+    Returns
+    -------
+    eigenvalues : :obj:`list` of :obj:`float`
+        The eigenvalues of the principal components.
+    scores : :class:`.BlockMatrix`
+        The principal component scores.
+    loadings : :class:`.Table`
+        The principal component loadings.
+    """
+    partition_table = _partition_samples(genotypes, relatedness_threshold, divergence_threshold)
+    unrelated_table = partition_table.filter(partition_table.is_in_unrelated)
+    matrix_table = matrix_table_source('pc_air/genotypes', genotypes)
+    field_name = matrix_table._fields_inverse[genotypes]
+    unrelated_genotypes = matrix_table.semi_join_cols(unrelated_table)[field_name]
+    eigenvalues, _, loadings = hl._hwe_normalized_blanczos(unrelated_genotypes, compute_loadings=True)
+    _loadings = np.array(loadings.loadings.collect())
+    standardized_genotypes = hwe_normalize(genotypes)
+    standardized_genotypes = BlockMatrix.from_entry_expr(standardized_genotypes).T
+    scores = standardized_genotypes @ _loadings
+    return eigenvalues, scores, loadings
```

Comment:
I want to make sure I got the essence of this method correct here. With this approach, I do not modify any of the existing methods.

If this is essentially correct, then I think all that's left to do is determine the best return type to use and perhaps avoid the redundant standardization. (Both `_hwe_normalized_blanczos` and `hwe_normalize` standardize the genotypes.)

For the return types, should I match the format `_hwe_normalized_blanczos` uses, returning the scores and the loadings as tables with one row field keyed by the sample key? (I think this is what @konradjk suggested as well.)